haddock-api/src/Haddock/Interface/Rename.hs

```diff
@@ -46,69 +45,60 @@ import GHC.Types.Basic ( TopLevelFlag(..) )
 --
 -- The renamed output gets written into fields in the Haddock interface record
 -- that were previously left empty.
-renameInterface :: DynFlags -> [String] -> LinkEnv -> Bool -> Interface -> ErrMsgM Interface
-renameInterface _dflags ignoredSymbols renamingEnv warnings iface =
+renameInterface :: ReportErrorMessage m => DynFlags -> Set String -> LinkEnv -> Bool -> Interface -> m Interface
```

Comment:
The `[String]` was being converted into a `Set String` at each iteration. I pushed that out so it would be faster.

The code is also now generic over the `ReportErrorMessage` class. In practice, it is only instantiated at `Ghc`.